nixos/modules/services/security/nginx-sso.nix

```
@@ -47,14 +51,34 @@ in {
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       serviceConfig = {
+        StateDirectory = "nginx-sso";
+        WorkingDirectory = "/var/lib/nginx-sso";
+        ExecStartPre = ''+${pkgs.writeShellScript "merge-nginx-sso-config" ''
+          rm -f '${configPath}'
+          # Relies on YAML being a superset of JSON
+          ${utils.genJqSecretsReplacementSnippet cfg.configuration configPath}
+
+          # Fix permissions
+          chown nginx-sso:nginx-sso ${configPath}
+          chmod 0600 ${configPath}
+        ''
+        }'';
```

```suggestion
        ExecStartPre = pkgs.writeShellScript "merge-nginx-sso-config" ''
          rm -f '${configPath}'
          # Relies on YAML being a superset of JSON
          ${utils.genJqSecretsReplacementSnippet cfg.configuration configPath}
          # Fix permissions
          chown nginx-sso:nginx-sso ${configPath}
          chmod 0600 ${configPath}
        '';
```
Fixing permissions afterwards leaves us with a race condition where the permissions are not correct.

---

The `+` in front of the script is load-bearing, it means it is executing as `root`.

---

Added a note about the files needing to be readable by `nginx-sso`. We can remove the `+`.